apps/arena/lib/arena/matchmaking/deathmatch_mode.ex

```diff
@@ -60,19 +58,43 @@ defmodule Arena.Matchmaking.DeathmatchMode do
     Process.send_after(self(), :launch_game?, 300)
     diff = System.monotonic_time(:millisecond) - state.batch_start_at
 
-    if length(clients) >= Application.get_env(:arena, :players_needed_in_match) or
-         (diff >= Utils.start_timeout_ms() and length(clients) > 0) do
+    state =
+      if Map.has_key?(state, :game_mode_configuration) do
+        state
+      else
+        case Arena.Configuration.get_game_mode_configuration("deathmatch", "deathmatch") do
+          {:error, _} ->
+            state
+
+          {:ok, game_mode_configuration} ->
+            # This is needed because we might not want to send a request every 300 seconds to the game backend
+            Process.send_after(self(), :update_params, 5000)
```

Comment:
This function is called every 300 milliseconds.
If we fall here every 300 ms, we are queueing a message for 5 seconds from here, but every 300 ms, so we're just delaying the problem.
If we want to just update the params every 5 seconds, we should take this out of this function.